### ISSUE TITLE
Build fails on Centos7/manylinux2014 and Ubuntu:latest

### ISSUE THREAD

**Issue by hogepodge** [ARM-software/ethos-n-driver-stack#3]
The build on Centos7/manylinux2014 fails at build stage:

`g++ -o support_library/build/release/src/Support.o -c -std=c++14 -O3 -Werror -Wall -Wextra -Wformat=2 -Wno-format-nonliteral -Wctor-dtor-privacy -Woverloaded-virtual -Wsign-promo -Wstrict-overflow=2 -Wswitch-default -Wlogical-op -Wnoexcept -Wstrict-null-sentinel -Wconversion -fPIC -Icommand_stream/include -Iutils/include -Isupport_library/build/release/src -Isupport_library/src -Isupport_library/build/release/include -Isupport_library/include support_library/src/Support.cpp`

With warnings that include:

`In file included from support_library/src/Graph.hpp:9,
                  from support_library/src/Compiler.hpp:10,
                  from support_library/src/Support.cpp:9:
 /opt/rh/devtoolset-9/root/usr/include/c++/9/ext/new_allocator.h: In instantiation of 'void __gnu_cxx::new_allocator<_Tp>::construct(_Up*, _Args&& ...) [with _Up = ethosn::support_library::Operand::Consumer; _Args = {ethosn::support_library::Operation&, const long unsigned int&}; _Tp = ethosn::support_library::Operand::Consumer]':
 /opt/rh/devtoolset-9/root/usr/include/c++/9/bits/alloc_traits.h:482:2:   required from 'static void std::allocator_traits<std::allocator<_CharT> >::construct(std::allocator_traits<std::allocator<_CharT> >::allocator_type&, _Up*, _Args&& ...) [with _Up = ethosn::support_library::Operand::Consumer; _Args = {ethosn::support_library::Operation&, const long unsigned int&}; _Tp = ethosn::support_library::Operand::Consumer; std::allocator_traits<std::allocator<_CharT> >::allocator_type = std::allocator<ethosn::support_library::Operand::Consumer>]'
 /opt/rh/devtoolset-9/root/usr/include/c++/9/bits/vector.tcc:115:30:   required from 'void std::vector<_Tp, _Alloc>::emplace_back(_Args&& ...) [with _Args = {ethosn::support_library::Operation&, const long unsigned int&}; _Tp = ethosn::support_library::Operand::Consumer; _Alloc = std::allocator<ethosn::support_library::Operand::Consumer>]'
support_library/src/Network.hpp:46:50:   required from here
support_library/src/Network.hpp:28:19: error: but 'constexpr ethosn::support_library::Operand::Consumer::Consumer(ethosn::support_library::Operation&, size_t)' does not throw; perhaps it should be declared 'noexcept' [-Werror=noexcept]
    28 |         constexpr Consumer(Operation& operation, const size_t inputIndex)
       |                   ^~~~~~~~
`

Resulting in 

`cc1plus: all warnings being treated as errors
 scons: *** [support_library/build/release/src/Support.o] Error 1
 scons: building terminated because of errors.
`

Similar warnings appear regardless of gcc toolchain version installed (7 or 8).

**Comment by hogepodge** [comment]
Similarly in Ubuntu:

`support_library/src/nonCascading/Strategies.cpp: In function 'ethosn::support_library::{anonymous}::TryStripeShapesResult ethosn::support_library::{anonymous}::TryStripeShapes(const ethosn::support_library::SramAllocator&, const TensorShape&, const TensorShape&, const TensorShape&, const TensorShape&, ethosn::support_library::DataFormat, const TensorShape&, const ethosn::support_library::HardwareCapabilities&, const ethosn::support_library::utils::ShapeMultiplier&, const ethosn::support_library::utils::ShapeMultiplier&, std::pair<bool, unsigned int>, uint32_t, uint32_t, uint32_t)':
support_library/src/nonCascading/Strategies.cpp:368:27: error: noexcept-expression evaluates to 'false' because of a call to 'ethosn::support_library::InputStats::InputStats()' [-Werror=noexcept]
  368 |     TryStripeShapesResult result;
      |                           ^~~~~~
In file included from support_library/src/DebuggingContext.hpp:8,
                 from support_library/src/Graph.hpp:8,
                 from support_library/src/GraphNodes.hpp:8,
                 from support_library/src/nonCascading/Strategies.hpp:8,
                 from support_library/src/nonCascading/Strategies.cpp:6:
support_library/src/../include/ethosn_support_library/Support.hpp:227:5: error: but 'ethosn::support_library::InputStats::InputStats()' does not throw; perhaps it should be declared 'noexcept' [-Werror=noexcept]
  227 |     InputStats()
      |     ^~~~~~~~~~
cc1plus: all warnings being treated as errors
scons: *** [support_library/build/release_native/src/nonCascading/Strategies.o] Error 1
scons: building terminated because of errors.
`

**Comment by jslater8** [comment]
Thank you for raising this issue. We test with Ubuntu 18.04 LTS. Which tag/commit of the ethos-n-driver-stack are you trying to compile? 

Thanks,
Josh

**Comment by hogepodge** [comment]
I'm building in Docker containers, in this case 20.04 and gcc 9.3.0. It appears that the larger issue is more recent compilers have a wider set of warnings that they will emit, and it's a policy in the build file to fail on all warnings. This is preventing compilation and distribution with projects like TVM.

Still in one instance I attempted to pin back to an older version of GCC, 7.4 in one instance, and still ran into this error. The only environment I've been able to build unmodified in is the NVidia-cu100 container.
